opal/corelib/runtime.js

```diff
@@ -1321,7 +1321,7 @@
       // A splatted array must be copied
       return value.slice();
     }
-    else if (value['$respond_to?']('to_a', true)) {
+    else if (typeof(value.$to_a) === 'function' && !value.$to_a.$$stub) {
```

Comment:
Maybe the solution is to add `respond_to?` to the smaller Opal builds, then. If it's a dependency of basic syntax, it should be available regardless of how the runtime is loaded.